beacon-chain/db/kv/genesis.go

```diff
@@ -0,0 +1,123 @@
+package kv
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/pkg/errors"
+	eth "github.com/prysmaticlabs/ethereumapis/eth/v1alpha1"
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/blocks"
+	dbIface "github.com/prysmaticlabs/prysm/beacon-chain/db/iface"
+	iface "github.com/prysmaticlabs/prysm/beacon-chain/state/interface"
+	state "github.com/prysmaticlabs/prysm/beacon-chain/state/stateV0"
+	pbp2p "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
+	"github.com/prysmaticlabs/prysm/shared/params"
+)
+
+// SaveGenesisData bootstraps the beaconDB with a given genesis state.
+func (s *Store) SaveGenesisData(ctx context.Context, genesisState iface.BeaconState) error {
+	stateRoot, err := genesisState.HashTreeRoot(ctx)
+	if err != nil {
+		return err
+	}
+	genesisBlk := blocks.NewGenesisBlock(stateRoot[:])
+	genesisBlkRoot, err := genesisBlk.Block.HashTreeRoot()
+	if err != nil {
+		return errors.Wrap(err, "could not get genesis block root")
+	}
+	if err := s.SaveBlock(ctx, genesisBlk); err != nil {
+		return errors.Wrap(err, "could not save genesis block")
+	}
+	if err := s.SaveState(ctx, genesisState, genesisBlkRoot); err != nil {
+		return errors.Wrap(err, "could not save genesis state")
+	}
+	if err := s.SaveStateSummary(ctx, &pbp2p.StateSummary{
+		Slot: 0,
+		Root: genesisBlkRoot[:],
+	}); err != nil {
+		return err
+	}
+
+	if err := s.SaveHeadBlockRoot(ctx, genesisBlkRoot); err != nil {
+		return errors.Wrap(err, "could not save head block root")
+	}
+	if err := s.SaveGenesisBlockRoot(ctx, genesisBlkRoot); err != nil {
+		return errors.Wrap(err, "could not save genesis block root")
+	}
+
+	genesisCheckpoint := &eth.Checkpoint{Root: genesisBlkRoot[:]}
```

Comment:
We need to be careful here. There's two types of genesis checkpoint:
1. `&eth.Checkpoint{Root: genesisBlkRoot[:]}`
2. `&eth.Checkpoint{Root: params.BeaconConfig().ZeroHash}`

I forgot which is used for peering (i think 2?) 

cc @nisdas for feedback

Reply:
I think this was meant to be genesisState.FinalizedCheckpoint()

https://github.com/prysmaticlabs/prysm/blob/beadec32b8a6260f6c5f91603d159c2d0e22b19e/beacon-chain/blockchain/service.go#L373

Reply:
This looks like a disparity between interop and blockchain production routine. I will use the production path instead of the interop path.

Reply:
Actually, looks like i can remove all of this checkpoint stuff. It wasn't used in production, only in interop for some reason